src/dream_oauth2.mli

```diff
@@ -1,22 +1,52 @@
 module User_profile : sig
   type t = {
-    user : string;
-    email : string;
+    id : string;
+    display_name : string;
+    email : string option;
+    provider : string;
   }
   (** Information about an authenticated user. *)
 end
 
-val signin_url :
-  ?valid_for:float ->
+type oauth2
+(** Values of this type represend configured OAuth2 providers. *)
```

Comment:
Technically, these are OAuth2 clients rather than providers. The `provider` field in the profile is named correctly, but these values here represent clients for those providers.